backend/src/controller/listing.ts

```diff
@@ -3,46 +3,139 @@ import ExampleListingList from "../constants/ExampleListingList";
+      productDescription: listingItem.productDescription,
+      productID: listingItem.productId,
+    });
   });
-});
+}
+fillFullInfo();
 
 module.exports.getAllListings = (req: Request, res: Response) => {
-  res.json(ExampleListingList);
+  ExampleListingList
+    ? res.json(ExampleListingList)
+    : res.status(400).json({ errorMessage: "Rosources not found" });
 };
 
 module.exports.getListing = (req: Request, res: Response) => {
   const listing = ExampleListingList.find(
     (Listing) => Listing.listingID === req.params.id
   );
-  listing ? res.json(listing) : res.send("listing doesn`t exist");
+  listing
+    ? res.json(listing)
+    : res.status(400).json({ errorMessage: "Rosources not found" });
 };
 
 module.exports.getFullInfoListings = (req: Request, res: Response) => {
-  res.json(fullInfoListings);
+  fillFullInfo();
+  fullInfoListings
+    ? res.json(fullInfoListings)
+    : res.status(400).json({ errorMessage: "Rosources not found" });
 };
 
 module.exports.getFullInfoListing = (req: Request, res: Response) => {
+  fillFullInfo();
   const fullInfoListing = fullInfoListings.find(
     (Listing) => Listing.listingID === req.params.id
   );
   fullInfoListing
     ? res.json(fullInfoListing)
-    : res.send("listing doesn`t exist");
+    : res.status(400).json({ errorMessage: "Rosources not found" });
+};
+
+module.exports.addListing = (req: Request, res: Response) => {
+  if (typeof req.body == undefined) {
+    res.status(400).json({ errorMessage: "Request body is undefined" });
+  } else {
+    if (
+      req.body.userID &&
+      req.body.productID &&
+      req.body.amount &&
+      req.body.duration
+    ) {
+      const listing = {
+        listingID: `listing_num_${Math.random() * 999999999999999}`,
+        userID: req.body.userID,
+        productID: req.body.productID,
+        price: req.body.price,
+        amount: req.body.amount,
+        highestBid: 0,
+        startDate: new Date(),
+        endDate: new Date(
+          new Date().setHours(new Date().getHours() + req.body.duration)
+        ),
+        likes: 0,
+      };
+      listingList.push(listing);
+      fillFullInfo();
+    } else {
+      res.status(400).json({ errorMessage: "Data has missing properties" });
+    }
+  }
+};
+
+module.exports.deleteListing = (req: Request, res: Response) => {
+  if (typeof req.body == undefined) {
+    res.status(400).json({ errorMessage: "Rosources not found" });
+  } else {
+    let index = listingList
+      .map(function (e) {
+        return e.listingID;
+      })
+      .indexOf(req.body.listingID);
+
+    index > -1
+      ? listingList.splice(index, 1)
+      : res.status(404).json({ errorMessage: "Rosources not found" });
+    fillFullInfo();
+  }
+};
+
+module.exports.editListing = (req: Request, res: Response) => {
+  if (typeof req.body == undefined) {
+    res.status(400).json({ errorMessage: "Rosources not found" });
+  } else {
+    let index = listingList
+      .map(function (e) {
```

Comment:
suggestion: if you need to find an index of an item, you can use the `findIndex()` method. It works like `.find()` but returns a number (index) instead of the whole item